fs-v1-multipart.go

```diff
@@ -70,9 +70,8 @@ func (fs fsObjects) listMultipartUploads(bucket, prefix, keyMarker, uploadIDMark
 		walkResultCh, endWalkCh := fs.listPool.Release(listParams{minioMetaBucket, recursive, multipartMarkerPath, multipartPrefixPath})
 		if walkResultCh == nil {
 			endWalkCh = make(chan struct{})
-			walkResultCh = fs.startTreeWalk(minioMetaBucket, multipartPrefixPath, multipartMarkerPath, recursive, func(bucket, object string) bool {
-				return fs.isMultipartUpload(bucket, object)
-			}, endWalkCh)
+			listDir := getListDir(fs.isMultipartUpload, fs.storage)
```

Comment:
getListDir() should be named appropriately here, perhaps like a factory or something. 

listDirFactory() would make sense it is returning a listDirFn